### PR TITLE
[i18n] Added German translations for default system channel types

### PR DESCRIPTION
**Pull request by cweitkamp**
- Added German translations for default system channel types
- Streamlined common aspects for i18n

Depends on #5895 

Signed-off-by: Christoph Weitkamp <github@christophweitkamp.de>

### CODE REVIEW

File: bundles/core/org.eclipse.smarthome.core.thing/ESH-INF/i18n/DefaultSystemChannels_de.properties [hunk 1]
```diff
@@ -1,7 +1,32 @@
...
+channel-type.system.signal-strength.state.option.2 = Durchschnittlich
+channel-type.system.signal-strength.state.option.3 = Gut
+channel-type.system.signal-strength.state.option.4 = Ausgezeichnet
 channel-type.system.low-battery.label = Niedriger Batteriestatus
 channel-type.system.battery-level.label = Batterieladung
 channel-type.system.trigger.label = Ausl�ser
 channel-type.system.rawbutton.label = Einfache Taste
 channel-type.system.button.label = Taste
 channel-type.system.rawrocker.label = Wipptaster
+channel-type.system.power.label = Power
```
[the remark this answers is not in the document]
Comment (maggu2810) [reply bundles/core/org.eclipse.smarthome.core.thing/ESH-INF/i18n/DefaultSystemChannels_de.properties]: Sure, you are right. I personally don't care for "Power", "Mute", etc. But we are surely to much technical and if a good translation is found it should be used.

[the remark this answers is not in the document]
Comment (sjka) [reply bundles/core/org.eclipse.smarthome.core.thing/ESH-INF/i18n/DefaultSystemChannels_de.properties]: How about "Netzschalter"?

**Comment by cweitkamp** [comment]
> Since you have touched the test case, could you please add tests for the other newly added labels too?

@triller-telekom Sure. But for which purpose? To make sure the default channel types are available? Or to make sure they are translated properly?

**Comment by triller-telekom** [comment]
> Or to make sure they are translated properly?

Since this is what is tested with the battery channel for example.

File: bundles/core/org.eclipse.smarthome.core.thing/ESH-INF/i18n/DefaultSystemChannels_de.properties (hunk 1 above)
[the remark this answers is not in the document]
Comment (cweitkamp) [reply bundles/core/org.eclipse.smarthome.core.thing/ESH-INF/i18n/DefaultSystemChannels_de.properties]: I second @maggu2810. Personally I am fine with it.

What about "Power-Taste" or "Ein- / Ausschalter"

**Comment by cweitkamp** [comment]
Okay, the translation feature.

But wouldn't it make sense to test for the existence of all default system channel types anyway. To make sure they are available in the framework and to prevent to step into a regression by accident?

File: bundles/core/org.eclipse.smarthome.core.thing/ESH-INF/i18n/DefaultSystemChannels_de.properties [hunk 2]
```diff
@@ -1,7 +1,32 @@
...
+channel-type.system.location.label = Ort
```
Comment (cweitkamp) [inline bundles/core/org.eclipse.smarthome.core.thing/ESH-INF/i18n/DefaultSystemChannels_de.properties]: `system.location` has a description, too.

File: bundles/core/org.eclipse.smarthome.core.thing/ESH-INF/i18n/DefaultSystemChannels_de.properties [hunk 3]
```diff
@@ -1,7 +1,32 @@
...
+channel-type.system.motion.label = Bewegung
```
Comment (cweitkamp) [inline bundles/core/org.eclipse.smarthome.core.thing/ESH-INF/i18n/DefaultSystemChannels_de.properties]: `system.motion` has a description, too.

**Comment by triller-telekom** [comment]
> But wouldn't it make sense to test for the existence of all default system channel types anyway. To make sure they are available in the framework and to prevent to step into a regression by accident?

Sure it makes sense. feel free to add them.

**Comment by cweitkamp** [comment]
I expect travis to fail because of https://github.com/eclipse/smarthome/pull/5855#issuecomment-404187310.

**Comment by triller-telekom** [comment]
Thanks for creating the tests and noticing that I missed to add the channels to the provider list. I have created https://github.com/eclipse/smarthome/pull/5895 to fix this.

File: bundles/core/org.eclipse.smarthome.core.thing/ESH-INF/i18n/DefaultSystemChannels_de.properties (hunk 1 above)
[the remark this answers is not in the document]
Comment (cweitkamp) [reply bundles/core/org.eclipse.smarthome.core.thing/ESH-INF/i18n/DefaultSystemChannels_de.properties]: Who will make the final decision?

@Confectrician May I ask for your opinion?

File: bundles/core/org.eclipse.smarthome.core.thing/ESH-INF/i18n/DefaultSystemChannels_de.properties [hunk 4]
```diff
@@ -1,7 +1,34 @@
...
+channel-type.system.signal-strength.state.option.4 = Ausgezeichnet
 channel-type.system.low-battery.label = Niedriger Batteriestatus
 channel-type.system.battery-level.label = Batterieladung
 channel-type.system.trigger.label = Ausl�ser
 channel-type.system.rawbutton.label = Einfache Taste
 channel-type.system.button.label = Taste
 channel-type.system.rawrocker.label = Wipptaster
+channel-type.system.power.label = Power
+channel-type.system.location.label = Ort
+channel-type.system.location.description = Ort in geographischen Koordinaten (Latitude/Longitude/Altitude).
```
Comment (NorbertHD) [inline bundles/core/org.eclipse.smarthome.core.thing/ESH-INF/i18n/DefaultSystemChannels_de.properties]: (Latitude/Longitude/Altitude) is english. Please change to (Breitengrad/Längengrad/Höhe).

File: bundles/core/org.eclipse.smarthome.core.thing/ESH-INF/i18n/DefaultSystemChannels_de.properties [hunk 5]
```diff
@@ -1,7 +1,34 @@
...
+channel-type.system.signal-strength.state.option.2 = Durchschnittlich
+channel-type.system.signal-strength.state.option.3 = Gut
+channel-type.system.signal-strength.state.option.4 = Ausgezeichnet
 channel-type.system.low-battery.label = Niedriger Batteriestatus
 channel-type.system.battery-level.label = Batterieladung
 channel-type.system.trigger.label = Ausl�ser
 channel-type.system.rawbutton.label = Einfache Taste
 channel-type.system.button.label = Taste
 channel-type.system.rawrocker.label = Wipptaster
+channel-type.system.power.label = Power
```
Comment (NorbertHD) [inline bundles/core/org.eclipse.smarthome.core.thing/ESH-INF/i18n/DefaultSystemChannels_de.properties]: Please change Power to Stromschalter.

File: bundles/core/org.eclipse.smarthome.core.thing/ESH-INF/i18n/DefaultSystemChannels_de.properties (hunk 4 above)
Comment (cweitkamp) [reply bundles/core/org.eclipse.smarthome.core.thing/ESH-INF/i18n/DefaultSystemChannels_de.properties]: Done. Thanks.

**Comment by cweitkamp** [comment]
I have rebased the branch. Lets have a look what Travis will say.

One open item left: the translation for `channel-type.system.power.label = Power`.

**Comment by sjka** [comment]
> One open item left: the translation for channel-type.system.power.label = Power.

We now have suggestions like "Strom", "Hauptschalter", "Stromschalter" and "Netzschalter" (which I even proposed myself), but thinking about it they are all not correct. 

For one, we are technically not discussing an abstraction of the power _button_ here, but rather the working mode of the device (which by coincidence is switchable between two states for most devices, therefore it is a channel of type switch - although this assumption may not hold true for every device...). The button is just another user interface for it. So the "Schalter" in there doesn't really make sense.

Furthermore, "strom" or "netz" also don't apply here, as that is not what they are doing. Such a mechanism cuts off the device from power mechanically, so that it becomes impossible to communicate with it electronically. Therefore such a channel wouldn't make sense - the whole Thing goes OFFLINE instead. 

Technically speaking it would have to be something like "Betriebsmodus" which in our simplified view on the world can be "OFF" (aka "sleep"/"Energiesparmodus") and "ON" (aka fully operational). Sounds a bit weird, but I think it is more to the point than what we previously had.

The opposite would work as well, i.e. "Energiesparmodus" - the only downside is that inverse view on it, as if it is "ON", the device actually is "OFF" and vice versa. Not very intuitive and therefore not really feasible in this context here.

**Comment by htreu** [comment]
Maybe we could shorten "Betriebsmodus" (which I like for its technical correctness) to "Betrieb" with states `ON` & `OFF`? Indicating that the device is operable when "Betrieb" has state `ON`.

**Comment by cweitkamp** [comment]
Alright. If nobody disagrees I will change the translation to "Betrieb" and add a meaningful description.

**Comment by maggu2810** [comment]
All fine now?

**Comment by cweitkamp** [comment]
I have changed everything. From my side it is ready to be merged.

**Comment by maggu2810** [comment]
So, let's wait for Travis CI and thank you for your work.

**Comment by cweitkamp** [comment]
Travis succeeded. Thanks to all of you.

**Pull request merged**